pandas/core/indexing.py

```diff
@@ -611,6 +615,42 @@ def _get_setitem_indexer(self, key):
                 raise
             raise IndexingError(key) from e
 
+    def _ensure_listlike_indexer(self, key, axis=None):
+        """
+        Ensure that a list-like of column labels are all present by adding them if
+        they do not already exist.
+
+        Parameters
+        ----------
+        key : _LocIndexer key or list-like of column labels
+            Target labels.
+        axis : key axis if known
+        """
+        column_axis = 1
+
+        # column only exists in 2-dimensional DataFrame
+        if self.ndim != 2:
+            return
+
+        if isinstance(key, tuple):
+            # key may be a tuple if key is a _LocIndexer key
+            # in that case, set key to the column part of key
+            key = key[column_axis]
+            axis = column_axis
+
+        if (
+            axis == column_axis
+            and not isinstance(self.obj._get_axis(column_axis), ABCMultiIndex)
+            and is_list_like_indexer(key)
+            and not com.is_bool_indexer(key)
+            and all(is_hashable(k) for k in key)
```

Comment:
i think the hashable check you can skip and the calls below will take care of it

Reply:
If no hashable check is performed here, some test cases fail (eg. https://github.com/pandas-dev/pandas/blob/master/pandas/tests/indexing/test_indexing.py#L112)
 because different errors are raised compared to the original code.